L64-python-runtime-proto-parsing.md

```diff
@@ -0,0 +1,99 @@
+gRPC Python Runtime Proto Parsing
+----
+* Author(s): gnossen
+* Approver: lidizheng
+* Status: Draft
+* Implemented in: Python
+* Last updated: April 9, 2020
+* Discussion at: **TODO(gnossen)**
+
+## Abstract
+
+Generated protobuf code is hard to manage for those without monorepos equipped with a hermetic build system. This has been echoed by the maintainers of libraries wrapping gRPC as well as direct users of gRPC. In this document, we aim to lay out a more Pythonic way of dealing with Protocol Buffer-backed modules.
+
+
+## Background
+
+gRPC and protocol buffers evolved in a Google ecosystem supported by a [monorepo](https://www.wired.com/2015/09/google-2-billion-lines-codeand-one-place/) and [hermetic build system](https://arxiv.org/pdf/1702.01715.pdf). When engineers check a `.proto` file into source control using that toolchain, the resultant code generated by the protocol compiler does not end up in source control. Instead, the code is generated on demand during builds and cached in a distributed build artifact store. While such sophisticated machinery has started to become available in the open source community (e.g. [Bazel](https://bazel.build/)), it has not yet gained much traction. Instead, small Git repos and non-hermetic language-specific build tools are the norm. As a result, code generated by the protocol compiler is often checked into source control alongside the code which uses it.
```

Comment:
Should we mention that community are building tools to ease this process like https://github.com/uber/prototool.

Reply:
We could. But it's not exactly apropos of this proposal. It's certainly good information though. Perhaps this sort of thing would make more sense as part of a post on the blog?